File: insertion/insertion.py
import random
from itertools import repeat

def in_place(lista):
    for i in range(1, len(lista)):
        j = i
        while j > 0 and lista[j] < lista[j - 1]:
            lista[j], lista[j - 1] = lista[j - 1], lista[j]
            j -= 1


def out_place(lista):
    ordenado = []
    ordenado.append(lista[0])
    for i in range(1, len(lista)):
        ordenado.append(lista[i])
        j = i
        while j > 0 and ordenado[j] < ordenado[j - 1]:
            ordenado[j], ordenado[j - 1] = ordenado[j - 1], ordenado[j]
            j -= 1
    return ordenado

if __name__ == '__main__':
    lista1 = [random.randrange(100000) for _ in repeat(None, 100)] + [-100000]

    lista2 = out_place(lista1)
    print(lista2)

    in_place(lista1)
    print(lista1)

    assert lista1 == list(sorted(lista1)) == lista2

File: insertion/Insertion.java
import java.util.Arrays;
import java.util.Random;

public class Insertion {

    public static void main(String[] args) {
        Random random = new Random();
        int[] lista1 = random.ints(10000, 0, 100000).toArray();

        int[] ordenado = outPlace(lista1);
        System.out.println(Arrays.toString(ordenado));

        int[] lista2 = Arrays.copyOf(lista1, lista1.length);
        inPlace(lista2);
        System.out.println(Arrays.toString(lista2));

        System.out.println(Arrays.equals(ordenado, lista2));
    }

    private static void inPlace(int[] v) {
        for (int i = 0; i < v.length; i++) {
            int j = i;
            while (j > 0 && v[j] < v[j - 1]) {
                swap(v, j, j - 1);
                j--;
            }
        }
    }

    private static int[] outPlace(int[] original) {
        int[] ordenado = new int[original.length];
        ordenado[0] = original[0];

        for (int i = 0; i < original.length; i++) {
            ordenado[i] = original[i];
            int j = i;
            while (j > 0 && ordenado[j] < ordenado[j - 1]) {
                swap(ordenado, j, j - 1);
                j--;
            }
        }

        return ordenado;
    }

    private static void swap(int[] v, int i, int j) {
        int aux = v[i];
        v[i] = v[j];
        v[j] = aux;
    }
}